### PR TITLE
Use status, kill, start, and check for dmypy

### LINKED ISSUE #44: Two invocations of dmypy made and appear to invalidate the cache
I'm using pylsp-mypy with neovim native lsp in dmypy mode, and when I save a file I don't get any of the speedup from using dmypy, so it takes 10s of seconds for type checking to load. When I use dmypy on the command line individually it runs instantaneously after edits (outside of neovim). If I edit within neovim (and therefore run dmypy through this plugin), then the mypy cache appears to get entirely recalculated and no caching happens.

```lua
local lsp = require('lspconfig')
local util = require('lspconfig.util')
lsp.pylsp.setup({
  settings = {
    pylsp = {
      plugins = {
        pylsp_mypy = {
          live_mode = false,
          enabled = true,
          dmypy = true,
        },
      }
    },
  },
  root_dir = function(fname)
    local root_files = {
      'requirements.txt',
    }
    return util.root_pattern(unpack(root_files))(fname) or util.find_git_ancestor(fname)
  end,
})
```

After editing a file and running `ps aux | grep mypy` I get two identical looking commands as output:

```
.direnv/python-3.7.15/bin/python3.7 .direnv/python-3.7.15/bin/dmypy run -- --show-column-numbers <file>
```

```
❯ pip freeze | grep -E '(lsp|mypy)'
mypy==0.931
mypy-extensions==0.4.3
mypy-protobuf==3.2.0
pylsp-mypy==0.6.3
pylsp-rope==0.1.10
python-lsp-black==1.2.1
python-lsp-jsonrpc==1.0.0
python-lsp-server==1.5.0
```

### PR DESCRIPTION
**Pull request by tyoung-patreon**
Fixes usage of dmypy so only one instance is started and they don't constantly kill eachother.

Fixes #44

### CODE REVIEW

**Comment by Richardk2n** [comment]
Not calling status is something that changed some time ago with no particular reason as far as I can tell, so we can fix that.

Why call start after kill, would the next run not do that anyway?

Does this fix your issue?

**Comment by tyoung-patreon** [comment]
This change did fix my issues, I think run works fine as well but I definitely found it more confusing when debugging and trying to identify which process was the daemon and which was being submitted to the daemon.

The bad interaction happens in the following scenario:

1. pylsp-mypy runs and starts a new dmypy command, it starts processing
2. I open a new file
3. pylsp-mypy runs and checks if dmypy is running. Instead of a successful response it just hangs
4. pylsp-mypy kills the original process and starts a new one

This continues until I stop editing files or I wait long enough for one of the dmypy tasks to fully index. It's a pretty rough experience for my project where the initial dmypy run takes several minutes, and during that time if I edit anything it starts over entirely. When I was debugging this is was helpful to have a `dmypy start` command distinct from the `dmypy run` commands so I knew which process was which, but maybe that's just personal preference.

File: pylsp_mypy/plugin.py [hunk 1]
```diff
@@ -250,6 +250,7 @@ def pylsp_lint(
                     errors.strip(),
                 )
                 subprocess.run(["dmypy", "kill"], **windows_flag)
+                subprocess.run(["dmypy", "start"], **windows_flag)
```
Comment (StabbarN) [inline pylsp_mypy/plugin.py:253]: Nitpicking :)

```suggestion
                subprocess.run(["dmypy", "restart"], **windows_flag)
```
`dmypy --help` states
```
...
restart           Restart daemon (stop or kill followed by start)
...
```

File: pylsp_mypy/plugin.py [hunk 2]
```diff
@@ -261,6 +262,7 @@ def pylsp_lint(
                     errors.strip(),
                 )
                 mypy_api.run_dmypy(["kill"])
+                mypy_api.run_dmypy(["start"])
```
Comment (StabbarN) [inline pylsp_mypy/plugin.py:265]: Nitpicking :)

```suggestion
                mypy_api.run_dmypy(["restart"])
```
`dmypy --help` states
```
...
restart           Restart daemon (stop or kill followed by start)
...
```

**Review by StabbarN — CHANGES_REQUESTED** [review]

**Comment by delfick** [comment]
hello, I'm wondering if you have an idea of when you'll release a new version with this change? thanks!

**Comment by Richardk2n** [comment]
Now

**Comment by delfick** [comment]
perfect, thanks!

**Pull request merged**